### PR TITLE
diagnostics: fix 10409

### PR DESCRIPTION
**Pull request by dvovk**
Fixed https://github.com/ledgerwatch/erigon/issues/10490

### CODE REVIEW

File: erigon-lib/diagnostics/stages.go [hunk 1]
```diff
@@ -83,6 +83,10 @@ func ReadStagesList(db kv.RoDB) []string {
 	var info []string
 	err := json.Unmarshal(data, &info)
 
+	if len(data) == 0 {
+		return []string{}
```
Comment (awskii) [inline erigon-lib/diagnostics/stages.go:87]: you can do that checks straight after you fetched `data` which is preferable way. So, you got data, you validate if its empty, if not - call `Unmarshal` and check `err` after call

File: erigon-lib/diagnostics/snapshots.go [hunk 2]
```diff
@@ -303,6 +303,10 @@ func ReadSnapshotDownloadInfo(db kv.RoDB) (info SnapshotDownloadStatistics) {
 	data := ReadDataFromTable(db, kv.DiagSyncStages, SnapshotDownloadStatisticsKey)
 	err := json.Unmarshal(data, &info)
 
+	if len(data) == 0 {
```
Comment (awskii) [inline erigon-lib/diagnostics/snapshots.go:306]: same

File: erigon-lib/diagnostics/snapshots.go [hunk 3]
```diff
@@ -315,6 +319,10 @@ func ReadSnapshotIndexingInfo(db kv.RoDB) (info SnapshotIndexingStatistics) {
 	data := ReadDataFromTable(db, kv.DiagSyncStages, SnapshotIndexingStatisticsKey)
 	err := json.Unmarshal(data, &info)
 
+	if len(data) == 0 {
```
Comment (awskii) [inline erigon-lib/diagnostics/snapshots.go:322]: same

**Review by awskii — CHANGES_REQUESTED** [review]

File: erigon-lib/diagnostics/snapshots.go (hunk 3 above)
Comment (dvovk) [reply erigon-lib/diagnostics/snapshots.go:322]: done

File: erigon-lib/diagnostics/snapshots.go (hunk 2 above)
Comment (dvovk) [reply erigon-lib/diagnostics/snapshots.go:305]: done

**Pull request merged**